Add documentation for private clusters and VCN peering (#68)
oracle/cluster-api-provider-oci: docs/src/networking/private-cluster.md

# Using Private clusters

> Note: This section has to be used only if the CAPOCI manages the workload cluster VCN. If externally managed VCN is
> used, this section is not applicable.

## Example Spec for Private Cluster

> CAPOCI supports Private clusters where the Kubernetes API Server Endpoint is a private IP Address

Author: will use private clusters everywhere

  compartmentId: "${OCI_COMPARTMENT_ID}"
  networkSpec:
    vcn:
      subnets:
        - cidr: 10.1.0.8/29
          name: control-plane-endpoint
          role: control-plane-endpoint
          type: private
        - cidr: 10.1.0.0/29
          name: control-plane
          role: control-plane
          type: private
        - cidr: 10.1.0.32/27
          name: service-lb
          role: service-lb
          type: public
        - cidr: 10.1.64.0/20
          name: worker
          role: worker
          type: private
```

## Example spec for VCN Peering using Dynamic Routing Gateway(Local)

While using Private Clusters, the management cluster needs to talk to the workload cluster. If the
management cluster and workload cluster are in separate VCN, the VCN peering can be used to connect the management
and workload cluster VCNS. CAPOCI supports peering of the workload cluster VCN with another VCN in the same region using
[Dynamic Routing Gateway][drg].

In case of Local VCN peering, a DRG OCID has to be provided and CAPOCI will attach the workload cluster VCN to the
provided DRG. The recommendation is to attach the management cluster VCN also to the same DRG so that the VCNs are

> peered to each other. More details about Local VCN Peering using DRG can be found [here][drg-local].

Reviewer: ```suggestion
peered to each other. More details see [Local VCN Peering using Local Peering Gateways][drg-local].
```

> An example template for this cluster-template-local-vcn-peering.yaml can be found in the Assets section under the

Reviewer: ```suggestion
An example template for this `templates/cluster-template-local-vcn-peering.yaml` can be found in the Assets section under the
```

Since the file is called out? 🤷‍♂️

Author: done

 [CAPOCI release page][capi-latest-release]. In order to use the template, the  environment variable has to be

> set. 

Author: done

The following Cluster API parameters have to be set in addition to the common parameters explained [here][common].

| Parameter | Default Value | Description                                                     |
|-----------|---------------|-----------------------------------------------------------------|
| `DRG_ID`  |               | OCID of the DRG to which the worklaod cluster will be attached. |

> ## Example spec for VCN Peering using Dynamic Routing Gateway(Remote)

Reviewer: space between `Gateway (Remote)`

If the management cluster and workload cluster are in different OCI regions, then DRG can still be used. In this case,
in addition to VCN attachment, [Remote Peering Connection][drg-rpc] has to be used.

> In case of Remote VCN peering, a DRG will be created by CAPOCI, and thw workload cluster VCN will be attached to the

Reviewer: ```suggestion
In case of Remote VCN peering, a DRG will be created by CAPOCI, and the workload cluster VCN will be attached to the
```

DRG. In addition, a Remote DRG has to be provided. CAPOCI will create RPC in the local and remote VCN and the RPCs
connection will be established between the RPCs.

> An example template for this cluster-template-remote-vcn-peering.yaml can be found in the Assets section under the

Reviewer: ```suggestion
An example template for this `cluster-template-remote-vcn-peering.yaml` can be found in the Assets section under the
```
Since it is a file to make it stand out.

[CAPOCI release page][capi-latest-release]. 

> The following Cluster API parameters have to be set in addition to the common parameters explained [here][common].

Reviewer: ```suggestion
The following Cluster API parameters have to be set in addition to the common parameters in the [Workload Cluster Parameters table][common].
```

where common is https://oracle.github.io/cluster-api-provider-oci/gs/create-workload-cluster.html#workload-cluster-parameters maybe
